### PR TITLE
feat: update samples to use 0.3.0 windows-drivers-rs crates

### PR DESCRIPTION
**Pull request by NateD-MSFT**
Update this repo to use an upcoming 0.3.0 release of [windows-drivers-rs](https://github.com/microsoft/windows-drivers-rs/), including the changes from https://github.com/microsoft/windows-drivers-rs/pull/143.

This of course does not build today, but was validated with local changes.

### CODE REVIEW

**Comment by wmmc88** [comment]
@NateD-MSFT I finished this pr with the last couple of changes needed since 0.3.0 crates are out. please review :)

File: Cargo.toml [hunk 1]
```diff
@@ -12,17 +12,24 @@ publish = false
 repository = "https://github.com/microsoft/windows-rust-driver-samples"
 license = "MIT OR Apache-2.0"
 
+[workspace.metadata.wdk.driver-model]
+driver-type = "KMDF"
+kmdf-version-major = 1
+target-kmdf-version-minor = 33
+
+
```
Comment (wmmc88) [inline Cargo.toml:20]:
```suggestion
```

**Review by NateD-MSFT — COMMENTED** [review]
I can't formally mark approved since I started this PR (and I think because I committed your most recent suggestion) but LGTM.

**Review by wmmc88 — APPROVED** [review]

**Review by svasista-ms — APPROVED** [review]

**Comment by svasista-ms** [comment]
[celebrate]     Shravan Vasista reacted to your message:
________________________________

**Pull request merged**